core/src/main/java/org/jdbi/v3/core/util/bean/SnakeCaseColumnNameStrategy.java

```diff
@@ -30,23 +30,7 @@ public SnakeCaseColumnNameStrategy(Locale locale) {
 
     @Override
     public boolean nameMatches(String propertyName, String sqlColumnName) {
-        // Convert the property name from camel-case to underscores syntax. Freely adapted from Spring
-        // BeanPropertyRowMapper.
-        StringBuilder propertyNameWithUnderscores = new StringBuilder();
-        propertyNameWithUnderscores.append(propertyName.substring(0, 1));
-        for (int i = 1; i < propertyName.length(); i++) {
-            // Do case comparison using strings rather than chars (avoid to deal with non-BMP char handling).
-            String s = propertyName.substring(i, i + 1);
-            String slc = s.toLowerCase(locale);
-            if (!s.equals(slc)) {
-                // Different cases: tokenize.
-                propertyNameWithUnderscores.append("_").append(slc);
-            }
-            else {
-                propertyNameWithUnderscores.append(s);
-            }
-        }
-        return propertyNameWithUnderscores.toString().equals(sqlColumnName.toLowerCase(locale));
+        return sqlColumnName.replace("_", "").toLowerCase(locale).equals(propertyName.toLowerCase(locale));
```

Comment:
Yeah, I'm waffling back and forth between the approaches.  The downside of `equalsIgnoreCase` is it relies on the default locale, so it is hard to specify behavior, it depends on the environment you run the code in.  But maybe that's OK?